rfc/003-ed25519-verification.md

```diff
@@ -0,0 +1,41 @@
+# RFC 003: Ed25519 Verification
+
+## Changelog
+
+- August 21, 2020: initialized
+
+## Author(s)
+
+- Marko (@marbar3778)
+
+## Context
+
+Ed25519 keys are the only supported key types for Tendermint validators currently. Tendermint-Go wraps the ed25519 key implementation from the go standard library. As more clients are implemented to communicate with the canonical Tendermint implementation (Tendermint-Go) different implementations of ed25519 will be used. Due to [RFC 8032](https://www.rfc-editor.org/rfc/rfc8032.html) not guaranteeing implementation compatibility, Tendermint clients must to come to an agreement of how to guarantee implementation compatibility. [Zcash](https://z.cash/) has multiple implementations of their client and have identified this as a problem as well. The team at Zcash has made a proposal to address this issue, [Zcash improvement proposal 215](https://zips.z.cash/zip-0215).
+
+## Proposal
+
+- Tendermint-Go would adopt [hdevalence/ed25519consensus](https://github.com/hdevalence/ed25519consensus).
+  - This library is implemented as an extension of the go standard library one.
+- Tendermint-rs would adopt [ed25519-zebra](https://github.com/ZcashFoundation/ed25519-zebra)
+  - related [issue](https://github.com/informalsystems/tendermint-rs/issues/355)
+
+As signature verification is one of the major bottlenecks of Tendermint-go, if ZIP 215 is adopted batch verification of signatures will be safe in consensus critical areas.
+
+## Status
+
+Proposed
+
+## Consequences
+
+### Positive
+
+- Batch verification
+- Signature verification across implementations
+
+### Negative
+
+### Neutral
+
```

Comment:
Should reflect that the library does not come with the eg. the std go crypto libs, even though Filipp'os reviewed it, it's in some sense a more external dependency. And if we only use it for verification (ie. signing is done using the default libs) then we have additional imports and to remember to use the right one for verification. Worth mentioning these consequences

 we'd be using some standard lib for most operations (signing) and only the zip215 lib for the verification.

Reply:
Added. I put it in a sub section in negative for Tendermint-go.